src/main/java/Command.java

```diff
@@ -0,0 +1,185 @@
+     */
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+}
+
+/**
+ * Represents the command to mark a task as done.
+ */
+class MarkCommand extends Command {
+    private int taskNumber;
+
+    public MarkCommand(int taskNumber) {
+        this.taskNumber = taskNumber;
+    }
+
+    @Override
+    public void execute(todoList toDo, Ui ui){
+        toDo.markTaskAsDone(taskNumber);
+        Task task = toDo.getTask(taskNumber);
+        System.out.println("Marked > " + task.getDescription() + " < as done!");
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+}
+
+/**
+ * Represents the command to unmark a task as not done.
+ */
+class UnmarkCommand extends Command {
+    private int taskNumber;
+
+    public UnmarkCommand(int taskNumber) {
+        this.taskNumber = taskNumber;
+    }
+
+    @Override
+    public void execute(todoList toDo, Ui ui){
+        toDo.markTaskAsNotDone(taskNumber);
+        Task task = toDo.getTask(taskNumber);
+        System.out.println("Unmarked > " + task.getDescription() + " < as done!");
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+}
+
+/**
+ * Represents the command to delete a task from the to-do list.
+ */
+class DeleteCommand extends Command {
+    private int taskNumber;
+
+    public DeleteCommand(int taskNumber) {
+        this.taskNumber = taskNumber;
+    }
+
+    @Override
+    public void execute(todoList toDo, Ui ui){
+        toDo.deleteTask(taskNumber);
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+}
+
+/**
+ * Represents the command to add a new task to the to-do list.
+ */
+class AddCommand extends Command {
+    private Task task;
+
+    public AddCommand(Task task) {
+        this.task = task;
+    }
+
+    @Override
+    public void execute(todoList toDo, Ui ui){
+        toDo.addTask(task);
+        System.out.println("Added > " + task + " < !");
+    }
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+}
+
+/**
+ * Represents the command to exit the application.
+ */
+class ExitCommand extends Command {
+    @Override
+    public void execute(todoList toDo, Ui ui){
+
+    }
+
+    @Override
+    public boolean isExit() {
+        return true;
+    }
+}
+
+/**
+ * Represents the command to show help information to the user.
+ */
+class HelpCommand extends Command {
+    @Override
+    public void execute(todoList toDo, Ui ui){
+        ui.showHelpMsg();
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+}
+
+class FindCommand extends Command {
+    private String keyword;
+
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    @Override
+    public void execute(todoList toDo, Ui ui) {
+        toDo.findTask(keyword);
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+}
```

Comment:
you can group the classes using packages; that may be better than writing all related classes in the same file